boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,935 @@
+    uint8_t   register_address,
+    uint16_t *register_value)
+{
+    return Io_Efuse_ReadRegister(
+        register_address, register_value, CSB_AUX1_AUX2_GPIO_Port,
+        CSB_AUX1_AUX2_Pin, &aux1_aux2_efuse);
+}
+
+static ExitCode Io_Efuse_WriteRegister(
+    uint8_t         register_address,
+    uint16_t        register_value,
+    GPIO_TypeDef *  ChipSelect_GPIO_Port,
+    uint16_t        ChipSelect_GPIO_Pin,
+    struct EfuseIo *e_fuse)
+{
+    ExitCode exit_code;
+    uint16_t command = 0x0000U;
+
+    // Place the register address into bits 10->13
+    command =
+        (uint16_t)((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    // Place register value to be written into bits 0->8
+    command = (uint16_t)(command | (register_value & EFUSE_SI_DATA_MASK));
+    // Invert watchdog bit state (Note: It is safe to do so even if the watchdog
+    // is disabled)
+    if (e_fuse->wdin_bit_to_set)
+    {
+        SET_BIT(command, WATCHDOG_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(command, WATCHDOG_BIT);
+    }
+    // Invert watchdog bit state for next write
+    e_fuse->wdin_bit_to_set = !e_fuse->wdin_bit_to_set;
+
+    // Compute and set/clear parity value
+    Io_Efuse_SetParityBit(&command);
+
+    exit_code = Io_Efuse_WriteToEfuse(
+        &command, ChipSelect_GPIO_Port, ChipSelect_GPIO_Pin);
+
+    return exit_code;
+}
+
+static ExitCode Io_Efuse_ReadRegister(
+    uint8_t         register_address,
+    uint16_t *      register_value,
+    GPIO_TypeDef *  ChipSelect_GPIO_Port,
+    uint16_t        ChipSelect_GPIO_Pin,
+    struct EfuseIo *e_fuse)
+{
+    ExitCode exit_code;
+    uint16_t command   = 0x0000U;
+    uint16_t tx_data[] = { 0x0000U };
+
+    // Place the Status Register address into bits 10->13
+    command =
+        (uint16_t)((SI_STATR_0_ADDR & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    // Shift bit 3 of the address (SOA3: the channel number) to the 13th bit
+    command = (uint16_t)(
+        command | ((register_address & SOA3_MASK) << EFUSE_ADDR_SHIFT) |
+        (register_address & (SOA2_MASK | SOA1_MASK | SOA0_MASK)));
+    // Invert watchdog bit state (Note: It is safe to do so even if the watchdog
+    // is disabled)
+    if (e_fuse->wdin_bit_to_set)
+    {
+        SET_BIT(command, WATCHDOG_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(command, WATCHDOG_BIT);
+    }
+    // Invert watchdog bit state for next write
+    e_fuse->wdin_bit_to_set = !e_fuse->wdin_bit_to_set;
+
+    // Compute and set/clear parity bit
+    Io_Efuse_SetParityBit(&command);
+
+    tx_data[0] = command;
+    exit_code  = Io_Efuse_ReadFromEfuse(
+        tx_data, register_value, ChipSelect_GPIO_Port, ChipSelect_GPIO_Pin);
+    // Only return register contents and clear bits 9->15
+    CLEAR_BIT(*register_value, ~EFUSE_SO_DATA_MASK);
+
+    return exit_code;
+}
+
+static void Io_Efuse_SetParityBit(uint16_t *spi_command)
+{
+    uint16_t spi_data = *spi_command;
+    uint8_t  parity_bit;
```

Comment:
Should this be a bool

```suggestion
    bool  parity_bit;
```

Reply:
Actually should probably declare this in the for loop

Reply:
That works too, just tested